src/tsserver/webServer.ts

```diff
@@ -0,0 +1,128 @@
+/*@internal*/
+namespace ts.server {
+    declare const addEventListener: any;
+    declare const postMessage: any;
+    declare const close: any;
+    declare const location: any;
+    declare const XMLHttpRequest: any;
+    declare const self: any;
+
+    const nullLogger: Logger = {
+        close: noop,
+        hasLevel: returnFalse,
+        loggingEnabled: returnFalse,
+        perftrc: noop,
+        info: noop,
+        msg: noop,
+        startGroup: noop,
+        endGroup: noop,
+        getLogFileName: returnUndefined,
+    };
+
+    let unknownServerMode: string | undefined;
+    function parseServerMode(): LanguageServiceMode | undefined {
+        const mode = findArgument("--serverMode");
+        if (mode !== undefined) {
+            switch (mode.toLowerCase()) {
+                case "partialsemantic":
+                    return LanguageServiceMode.PartialSemantic;
+                case "syntactic":
+                    return LanguageServiceMode.Syntactic;
+                default:
+                    unknownServerMode = mode;
+                    break;
+            }
+        }
+        // Webserver defaults to partial semantic mode
+        return hasArgument("--syntaxOnly") ? LanguageServiceMode.Syntactic : LanguageServiceMode.PartialSemantic;

```

Comment:
Does this new scenario need to support a deprecated command-line flag?

Reply:
Good point. No .. will remove that.